packages/frontend/app/dashboard/components/dashboard.tsx

```diff
@@ -1,91 +1,47 @@
-    content: "Our regular CO visit with brother xyz...",
-    date: "11/01/2024",
-  },
-  {
-    title: "CO Visit",
-    content: "Our regular CO visit with brother xyz...",
-    date: "11/01/2024",
-  },
-];
+import React from "react";
 
-const congInformationBoardData = [
-  {
-    title: "Announcement",
-    content: "This can be a file, link, or a short description",
-    date: "11/01/2024",
-  },
-  {
-    title: "Announcement",
-    content: "This can be a file, link, or a short description",
-    date: "11/01/2024",
-  },
-  {
-    title: "Announcement",
-    content: "This can be a file, link, or a short description",
-    date: "11/01/2024",
-  },
-];
+import { Spinner } from "@nextui-org/react";
+import axios from "axios";
 
-const dutiesData = [
-  {
-    title: "Microphone",
-    content: "Left Microphone, Support back door in case of emergencies",
-    date: "11/01/2024",
-  },
-  {
-    title: "Microphone",
-    content: "Left Microphone, Support back door in case of emergencies",
-    date: "11/01/2024",
-  },
-  {
-    title: "Microphone",
-    content: "Left Microphone, Support back door in case of emergencies",
-    date: "11/01/2024",
-  },
-];
+import { AdminDashboard } from "./admin-dashboard";
+import { UserDashboard } from "./user-dashboard";
+
+import { backendRoutes } from "@/lib/config";
+import { requestOptions } from "@/lib/request-options";
 
 function Dashboard() {
-  return (
-    <div className="flex justify-evenly gap-x-4 mx-4">
-      <AnimateCard delay={0}>
-        <UserInfoCard
-          userInfo={{
-            name: "John Doe",
-            email: "example@gmail.com",
-            congregation: "London, Uxbridge",
-          }}
-        />
-      </AnimateCard>
-      <AnimateCard delay={0.1}>
-        <MeetingDutiesCard dutiesData={dutiesData} />
-      </AnimateCard>
-      <AnimateCard delay={0.2}>
-        <InformationCard informationBoardData={congInformationBoardData} />
-      </AnimateCard>
-      <AnimateCard delay={0.3}>
-        <CongEventsCard congEventsData={congEventsData} />
-      </AnimateCard>
-      <AnimateCard delay={0.4}>
-        <PublicWitnessingCard />
-      </AnimateCard>
-    </div>
-  );
+  const [currentUser, setCurrentUser] = React.useState<any>();
+  const [isLoading, setIsLoading] = React.useState<boolean>(true);
+
+  // This gets the current user from the backend
+  // and then sets it to some local state
+  const getCurrentUser = async () => {
+    try {
+      const res = await axios.get(backendRoutes.user.me, requestOptions());
+
+      setCurrentUser({ ...res.data });
+    } catch (err) {
+      console.error(err);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  React.useEffect(() => {
+    getCurrentUser();
+  }, []);
+
+  if (isLoading) {
+    return <Spinner />;
+  } else {
+    return currentUser.type === "REGULAR" ? (
+      <UserDashboard currentUser={currentUser} />
+    ) : (
+      <AdminDashboard currentUser={currentUser} />
+    );
+  }
```

Comment:
Best to use a map from the user type to the corresponding component. Better for conditional rendering in this situation